osp_scraper/spiders/ecollege.py

```
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+
+import scrapy
+
+from ..spiders.CustomSpider import CustomSpider
+
+class ILCCSpider(CustomSpider):
+    name = "ecollege"
+
+    def start_requests(self):
+        start_url = "https://secure.ecollege.com/iowacc/index.learn?action=Catalog"
+
+        def get_terms_codes(response):
+            for option in response.css("form input[type='Radio']"):
+                code = option.css("input::attr(value)").extract_first()
+
+                if code:
+                    yield scrapy.FormRequest(
+                        start_url + "&SID=" + code,
+                        method="POST",
+                        formdata={
+                            "SID": code
+                        },
+                        callback=parse_results
+                    )
+
+        def parse_results(response):
+            for row in response.css('table+table tr'):
+                url = row.css('td:last-child a::attr(href)').extract_first()
+                if url:
```

You're right that the selector you were using was appropriate.  I wasn't paying attention to the anchor you were getting.

You're also right about Scrapy just throwing an error and moving on.  I had done some lite testing a while back that made me think that Scrapy wouldn't handle bad URLs well and scrapers would crash because of that.  I double-checked and that's not true (it also wouldn't really make much sense if it did given what Scrapy is intended to accomplish).

That being said, I think your `if` check is appropriate here because sometimes there aren't `a` tags in the cells you're looking in, but as a general approach it seems problematic.  If something on the website subtly changed, for instance, I think I would rather Scrapy error and make some noise than have a scraper silently skip downloads.